### PR TITLE
Schemaspy metadata

### PR DESCRIPTION
**Pull request by joshuashaffer**
Implements a few metadata functions so SchemaSpy mostly works. Issue #26

### CODE REVIEW

File: src/main/java/org/duckdb/DuckDBDatabaseMetaData.java [hunk 1]
```diff
@@ -171,27 +171,76 @@ public String getIdentifierQuoteString() throws SQLException {
...
+        return sb.toString();
     }
 
     @Override
     public String getSystemFunctions() throws SQLException {
-        throw new SQLFeatureNotSupportedException("getSystemFunctions");
+        Statement statement = conn.createStatement();
+        statement.closeOnCompletion();
+        ResultSet rs = statement.executeQuery(
+                "SELECT DISTINCT function_name FROM duckdb_functions() WHERE parameter_types[1] IS NULL");
```
Comment (Mause) [inline src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:223]: This condition doesn't quite work - probably need to exclude vararg functions somehow?

**Review by Mause — CHANGES_REQUESTED** [review]

File: src/main/java/org/duckdb/DuckDBDatabaseMetaData.java (hunk 1 above)
Comment (joshuashaffer) [reply src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:223]:
```sql 
select DISTINCT function_name from duckdb_functions() where length(parameter_types) = 0
```
??

Comment (joshuashaffer) [reply src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:223]: It's not used consistently??

```sql 
D select DISTINCT function_name, parameter_types, varargs from duckdb_functions() where parameter_types[1] is null;
```

|           function_name            |  parameter_types   | varargs |
|------------------------------------|--------------------|---------|
| map_from_entries                   | []                 | ANY     |
| list_median                        | [NULL]             |         |
| ... | ... | ... |

Sometimes, they have a type in the Vargas columns; other times, they have an explicit `NULL` as the first argument.

Comment (joshuashaffer) [reply src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:223]: @Mause is this ok solution given the above, given that vargargs aren't handled consistently?

Comment (Mause) [reply src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:223]: Sounds good for now yeah

**Review by Mause — APPROVED** [review]

**Comment by Mause** [comment]
Thanks!

**Pull request merged**